types/src/v2/error.rs

```diff
@@ -1,14 +1,36 @@
+use crate::v2::params::{Id, TwoPointZero};
+use serde::de::{Deserializer, MapAccess, Visitor};
+use serde::ser::{SerializeMap, Serializer};
+use serde::{Deserialize, Serialize};
+use serde_json::value::RawValue;
+use std::fmt;
 use thiserror::Error;
 
-/// Error.
-#[derive(Error, Debug)]
-pub enum RpcError {
-	/// Unknown error.
-	#[error("unknown rpc error")]
-	Unknown,
-	/// Invalid params in the RPC call.
-	#[error("invalid params")]
-	InvalidParams,
+/// [Failed JSON-RPC response object](https://www.jsonrpc.org/specification#response_object).
+#[derive(Serialize, Debug)]
+pub struct JsonRpcError<'a> {
+	/// JSON-RPC version.
+	pub jsonrpc: TwoPointZero,
+	/// Error.
+	pub error: ErrorCode,
```

Comment:
@maciejhirsz I removed your type https://github.com/paritytech/jsonrpsee/blob/master/types/src/v2/mod.rs#L85-#L90. ErrorCode should be as cheap I think and more readable